tests/src/main/java/org/hibernate/beanvalidation/tck/tests/valueextraction/unwrapping/ValueExtractionUnwrappingTest.java

```diff
@@ -0,0 +1,304 @@
+	public void validate_wrapped_value_if_value_extractor_unwraps_by_default_and_unwrapping_enabled_per_constraint_for_non_generic_container() {
+		Set<ConstraintViolation<IntegerWrapperWithForcedUnwrapping>> constraintViolations = getValidatorWithValueExtractors().validate( new IntegerWrapperWithForcedUnwrapping() );
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationOf( Min.class ).withProperty( "integerWrapper" )
+		);
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTMETADATA_CONSTRAINTDESCRIPTOR, id = "p")
+	public void validateUnwrappedValue_returns_correct_values() {
+		Validator validator = getValidatorWithValueExtractors();
+		ConstraintDescriptor<?> minConstraintDescriptor = validator.getConstraintsForClass( WrapperWithImplicitUnwrapping.class )
+				.getConstraintsForProperty( "integerWrapper" )
+				.getConstraintDescriptors()
+				.iterator().next();
+
+		assertEquals( minConstraintDescriptor.getAnnotation().annotationType(), Min.class );
+		assertEquals( minConstraintDescriptor.validateUnwrappedValue(), ValidateUnwrappedValue.DEFAULT );
+
+		minConstraintDescriptor = validator.getConstraintsForClass( WrapperWithDisabledUnwrapping.class )
+				.getConstraintsForProperty( "integerWrapper" )
+				.getConstraintDescriptors()
+				.iterator().next();
+
+		assertEquals( minConstraintDescriptor.getAnnotation().annotationType(), NotNull.class );
+		assertEquals( minConstraintDescriptor.validateUnwrappedValue(), ValidateUnwrappedValue.NO );
+
+		minConstraintDescriptor = validator.getConstraintsForClass( WrapperWithForcedUnwrapping.class )
+				.getConstraintsForProperty( "integerWrapper" )
+				.getConstraintDescriptors()
+				.iterator().next();
+
+		assertEquals( minConstraintDescriptor.getAnnotation().annotationType(), Min.class );
+		assertEquals( minConstraintDescriptor.validateUnwrappedValue(), ValidateUnwrappedValue.YES );
+	}
+
+	private class EntityWithSkipAndUnwrapAtTheSameTime {
+
+		@NotNull(payload = { Unwrapping.Unwrap.class, Unwrapping.Skip.class })
+		private final ValueHolder<Integer> integerHolder = new ValueHolder<>( 5 );
+	}
+
+	private class EntityWithNoContraintValidatorForUnwrappedValue {
+
+		// no constraint validator for the wrapped value
+		@Future(payload = { Unwrapping.Unwrap.class })
+		private final ValueHolder<Integer> integerHolder = new ValueHolder<>( 5 );
+	}
+
+	private class EntityWithExplicitUnwrapping {
+
+		@Min(value = 10, payload = { Unwrapping.Unwrap.class })
+		private final ValueHolder<Integer> integerHolder = new ValueHolder<>( 5 );
+	}
+
+	private class EntityWithNoUnwrapping {
+
+		@Null
+		private final ValueHolder<Integer> integerHolder = new ValueHolder<>( 5 );
+	}
+
+	private class WrapperWithImplicitUnwrapping {
+
+		@Min(10)
+		private final Wrapper<Integer> integerWrapper = new Wrapper<>( 5 );
+	}
+
+	private class WrapperWithDisabledUnwrapping {
+
+		@NotNull(payload = { Unwrapping.Skip.class })
```

Comment:
Hmmm, no, because the wrapped value is null so it would throw an error (keep in mind that the VE for Wrapper is `@UnwrapByDefault`).